### PR TITLE
ci: use dockerLogs step

### PR DESCRIPTION
**Pull request by v1v**
### What

It archives the docker container details for every stage that runs docker.

### How to use it

- Go to the `artifact/docker-info` folder
- Look for the stage that you'd like to go deeper with
- Open the file `docker-containers.txt`
- Look for the container ID
- `<id>.log` is the logs for the container with `id`
- `<id>-cmd.txt` is the docker container run parameters
- `<id>-inspect.json` is the output of the docker inspect <id> command

### Tests

image

image

(BO view)

image

### Issues

Closes https://github.com/elastic/apm-agent-rum-js/issues/806

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env groovy
 
-@Library('apm@current') _
+@Library('apm@test/docker-logs') _
```
Comment (v1v) [inline Jenkinsfile:3]:
```suggestion
@Library('apm@current') _
```

File: Jenkinsfile [hunk 2]
```diff
@@ -389,16 +389,22 @@ def runScript(Map args = [:]){
         sleep randomNumber(min: 5, max: 10)
         sh(label: 'Pull and build docker infra', script: '.ci/scripts/pull_and_build.sh')
       }
-      // Another retry in case there are any environmental issues
-      retry(3) {
-        sleep randomNumber(min: 5, max: 10)
-        if(env.MODE == 'saucelabs'){
-          withSaucelabsEnv(){
+      try {
```
[the remark this answers is not in the document]
Comment (v1v) [reply Jenkinsfile:392]: Alright, I'm afraid we cannot do the post-block though, the `runAllScopes` runs several scopes in a loop :(

**Comment by v1v** [comment]
> I'm worried about this one. Won't docker inspect will dump all environment details? I'm concerned that we might end up leaking sensitive data. Even if that isn't the case now, we might accidentally forget that this is in place in the future.

What do you suggest?

**Comment by cachedout** [comment]
> What do you suggest?

Sorry, I didn't see this comment earlier. I don't think we should ever do a `docker inspect` on a production CI job. I see that https://github.com/elastic/apm-pipeline-library/pull/580 removes this capability so I think we are on the same page here. 👍

**Comment by kuisathaverat** [comment]
>I'm worried about this one. Won't docker inspect will dump all environment details? I'm concerned that we might end up leaking sensitive data. Even if that isn't the case now, we might accidentally forget that this is in place in the future.

should not be secrets references in the docker-compose.yml, and the ones that can contain are related to the local containers and do not give access anywhere. In some cases, we need the inspect to see the status of the container when it died, we can remove the environment from the inspect output when we generate the file.

**Comment by cachedout** [comment]
> we can remove the environment from the inspect output when we generate the file.

Parsing the output and removing the `Envrionment` value would satisfy my concerns. 👍

**Comment by kuisathaverat** [comment]
this makes the thing

`docker inspect 60754aee9a26 |jq 'del(.[].Config.Env)'`

**Pull request merged**